graphql_compiler/tests/test_input_data.py

```diff
@@ -448,7 +449,6 @@ def complex_optional_variables():
     # The operands in the @filter directives originate from an optional block.
     graphql_input = '''{
         Animal {
-            name @filter(op_name: "=", value: ["$animal_name"])
```

Comment:
It'd be preferable to keep this if it's not too much work -- it ensures that the query is scheduled and executed in a predictable order. Otherwise, we simply don't know if this location runs first, or its `out_Animal_ParentOf` is the first one.

Reply:
Hmm, but if we start from a specific `Animal`, we aren't testing all possible behaviors of the `@optional` traversals and tagged values. That's why I removed the filter. It should be easy enough to put it back, but the results will be less interesting.

Reply:
I'd put a separate test case with this different query. Having the filter there vs not having it trigger very different query planning decisions from the database, so I don't think this is quite as equivalent as you suggest.